sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_kernels.asciidoc

```diff
@@ -323,6 +325,118 @@ Otherwise `value` is `false`.
 The helper trait `is_kernel_v` provides the value of `value`.
 |====
 
+=== New free functions to launch a kernel
+
+This extension adds the following helper which captures a kernel function
+address as a template parameter.
+
+[frame=all,grid=none,separator="@"]
+!====
+a@
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template<auto *Func>
+struct kernel_function_s {};
+
+template<auto *Func>
+inline constexpr kernel_function_s<Func> kernel_function;
+
+} // namespace sycl::ext::oneapi::experimental
+----
+!====
+
+It also adds the following free functions which launch a free function kernel.
+
+[frame=all,grid=none,separator="@"]
+!====
+a@
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <auto *Func, typename Args...>
+void single_task(queue q, kernel_function_s<Func> k, Args&&... args);
+
+template <auto *Func, typename Args...>
+void single_task(handler &h, kernel_function_s<Func> k, Args&&... args);
+
+} // namespace sycl::ext::oneapi::experimental
+----
+!====
+
+_Constraints_: Available only if `is_single_task_kernel_v<Func>` is `true`.
+Available only if `+std::is_invocable_v<decltype(Func), Args...>+` is `true`.
+
+_Effects_: Enqueues a kernel object to the `queue` or `handler` as a single task.
+Each value in the `args` pack is passed to the corresponding argument in
+`Func`, converting it to the argument's type if necessary.
```

Comment:
@rolandschulz I think I need your help here.  It seems like there should be some metaprogramming way to convert each value in the `args` pack to the type of the corresponding argument in `Func`, but I'm exactly sure how to do that.  Do you think this is possible?  When we were talking about this before (a while ago), we were expecting that this would be possible, but I don't think we ever created a POC.

We could instead require that the type of each value in the `args` pack *exactly matches* the type of the corresponding argument to `Func`.  However, this would make it easy to shoot yourself in the foot because the `is_invocable_v` constraint will allow parameter values that are not the same (but are implicitly convertible).